deno/wipe_appwrite_bucket/src/mod.ts

```diff
@@ -0,0 +1,54 @@
+import { sdk } from "./deps.ts";
+
+export default async function(req: any, res: any) {
+  if(
+    !req.variables['APPWRITE_FUNCTION_ENDPOINT'] || 
+    !req.variables['APPWRITE_FUNCTION_PROJECT_ID'] || 
+    !req.variables['APPWRITE_FUNCTION_API_KEY'] ||
+    !req.payload
+  ) {
+    return res.json({
+      success: false,
+      message: "Missing required environment variables or payload.",
+    });
+  }
+  let bucketId: string = '';
+  try {
+    const payload = JSON.parse(req.payload);
+    bucketId = payload.bucketId;
+  } catch(err) {
+    return res.json({
+      success: false,
+      message: "Payload is invalid.",
+    });
+  }
+  const client = new sdk.Client();
+  const storage = new sdk.Storage(client);
+  
+  client
+    .setEndpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])
+    .setProject(req.variables['APPWRITE_FUNCTION_PROJECT_ID'])
+    .setKey(req.variables['APPWRITE_FUNCTION_API_KEY'])
+  ;
+
+  try {
+    const limit = 25;
+    let offset = 0;
+    let listFiles;
+    do {
+      listFiles = await storage.listFiles(bucketId, [sdk.Query.limit(limit), sdk.Query.offset(offset)]);
+      for(const file of listFiles.files) {
+        await storage.deleteFile(bucketId, file.$id);
```

Comment:
There isn't a need to `await` every file delete. It would be a good tradeoff between performance and inundating the server if each "page" of files were all deleted concurrently before fetching the next "page".

Reply:
Nice catch, I removed it. Thanks :)

Reply:
You should be able to delete every page concurrently rather than awaiting each delete. You can do something like:

```
while (true) {
  result = storage.listFiles();
  if (no results) break;
  
  promises = []
  result.files.forEach(file) {
    promise.push(storage.deleteFile(file);
  })
  
  try {
    await Promise.all(promises);
  } catch (e) {
    // handle accordingly
  }
}
```

Reply:
Thanks a lot. I optimized the loop now :)